ts/parsers/slideParser.ts

```diff
@@ -1,14 +1,141 @@
 import * as format from "string-template";
 import { getAttributeByPath, ZipHandler } from "../helpers";
-import { GraphicFrameParser, PowerpointElementParser } from "./";
+import { GraphicFrameParser, PowerpointElementParser, SlideRelationsParser } from "./";
 
 export default class SlideParser {
-    public static async getSlideElements(PPTElementParser: PowerpointElementParser, slideNumber): Promise<any[]> {
+    public static async getSlideLayout(slideRelations) {
+        // Read relationship filename of the slide (Get slideLayoutXX.xml)
+        // @sldFileName: ppt/slides/slide1.xml
+        // @resName: ppt/slides/_rels/slide1.xml.rels
+        let relationshipArray = slideRelations["Relationships"]["Relationship"];
+        let layoutFilename = "";
+        if (relationshipArray.constructor === Array) {
+            for (const relationship of relationshipArray) {
+                if (
+                    relationship["$"]["Type"] ===
+                    "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout"
```

Comment:
```suggestion
                    relationship["$"]["Type"] ===
                        "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout"
```

Reply:
This schema can also be a const defined in consts file if there is one, or as a class const.

Reply:
made a constants file and moved it there.